aws/cloudformation/components/database.yml.erb

```diff
@@ -187,4 +161,42 @@
       TargetGroupName: default
       ConnectionPoolConfigurationInfo:
         ConnectionBorrowTimeout: 10
-<%end -%>
+<%   end -%>
+<% end -%>
+# Publish identifiers for database Resources provisioned in this template to Secrets, so the values can dynamically
+# populate CDO.* application configuration settings. Publish these Secrets even when `database == false` so that
+# these configuration settings correctly point to MySQL service running on localhost in those environments.
+# TODO: Transition these configuration settings to AWS Systems Manager Parameter Store.
+# TODO: Add Stack name to the Secret name to support multiple deployments with the same environment type.
+<%
+    if database
+      if rack_env?(:production)
+        db_cluster_id = "#{PRODUCTION_DB_CLUSTER_ID}"
+        db_writer_endpoint = 'PLACEHOLDER' # Set the value of this Secret manually in production to avoid disclosing this hostname.
+      else
+        db_cluster_id = '!Ref AuroraCluster'
+        db_writer_endpoint = '!GetAtt AuroraCluster.Endpoint.Address'
+      end
+      db_proxy_reporting_endpoint = '!GetAtt ReportingDBProxyEndpoint.Endpoint'
+    else
+      db_cluster_id = db_writer_endpoint = db_proxy_reporting_endpoint = 'localhost'
+    end
+-%>
+  DBClusterIDConfig:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Description: 'Publish Database Cluster ID to application configuration setting CDO.db_cluster_id'
+      Name: '<%=environment%>/cdo/db_cluster_id'
+      SecretString: <%=db_cluster_id%>
+  DBWriterEndpointConfig:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Description: 'Publish Database Cluster Writer Endpoint to application configuration setting CDO.db_writer_endpoint'
+      Name: '<%=environment%>/cdo/db_writer_endpoint'
+      SecretString: <%=db_writer_endpoint%>
+  DBProxyReportingEndpointConfig:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Description: 'Publish RDS Proxy Reporting Endpoint to the application configuration setting CDO.db_proxy_reporting_endpoint'
```

Comment:
This is outside the scope of your changes, but these Descriptions seem inappropriate, when simply "RDS Proxy Reporting Endpoint" would be sufficient. The current description describes why we're creating a Secret resource instead of what the secret represents.